### DISCUSSION TITLE
Margin timestamps for process log

### DISCUSSION THREAD

**Discussion by phil-s** [magit/magit#4920] [Ideas]
I somewhat frequently want to check _when_ I last performed some particular git command in Magit, and I then always wish the process buffer had timestamps.

I'm wondering if those could be added as an option in the right margin, similarly to the author date display in log buffers?

**Comment by tarsius** [comment]
I've implemented this in 85d0554f4f8f219e5e7aaa02e44c65a1eae01b30. Unfortunately I forgot to restart Emacs before pushing and had to immediately revert it again. This leads to an awful circular dependency. Reducing circular dependencies is a more important goal than adding this small feature. I'll bang on it some more, but if there is no easy way out, then this will have to wait.

**Comment by phil-s** [comment]
What are your feelings on wrapping `(defcustom magit-process-margin ...)` in `(with-eval-after-load 'magit-margin ...)` ?

It does mess up the source link (using 28.2), so that's not ideal:
`magit-process-margin is a variable defined in ‘magit-core.el’.`

That aside, it seems to circumvent the issue.

Or this hack?:

```el
(defun magit-process-margin--setter (symbol value)
  (with-eval-after-load 'magit-margin
    (magit-margin-set-variable 'magit-process-mode symbol value)))

(defcustom magit-process-margin '(nil "%H:%M:%S" 8)
  "..."
  :package-version '(magit . "4.0.0")
  :group 'magit-process
  :group 'magit-margin
  :set #'magit-process-margin--setter
  :type '(list (boolean :tag "Show margin initially")
               (string  :tag "Time format string")
               (number  :tag "Time format string width")))
```

**Reply by tarsius** [reply]
> What are your feelings on wrapping `(defcustom magit-process-margin ...)` in `(with-eval-after-load 'magit-margin ...)` ?

I very much don't want to do that. This isn't about suppressing warnings -- it's about actually avoiding circular dependencies. I have already removed many circular dependencies last summer, but the most serious instances remain. I don't want to make it harder to eventually tackle those as well, by adding a new layer of not so serious circular dependencies on top.

Instead I have added the option `magit-process-timestamp-format`, which you can customize to instead insert the timestamp in the section heading.
